nat46/modules/nat46-netdev.c

```diff
@@ -251,6 +251,47 @@ int nat46_configure(char *devname, char *buf) {
 	}
 }
 
+int nat46_remove(char *devname, char *buf) {
+	int ret = -1;
+	int buflen = 1024;
+	char config_remove[buflen];
+	struct net_device *dev;
+	nat46_instance_t *nat46;
+	nat46_instance_t *nat46_remove;
+	int result_rem;
+	int i;
+
+	if((dev = find_dev(devname)) == NULL ||
+	   (nat46 = netdev_nat46_instance(dev)) == NULL ||
+	   (nat46_remove = alloc_nat46_instance(1, NULL, -1, -1, -1)) == NULL) {
+		return ret;
+	}
+
+	if(nat46_set_ipair_config(nat46_remove, 0, buf, buflen) < 0) {
+		release_nat46_instance(nat46_remove);
+		return ret;
+	}
+
+	result_rem = nat46_get_ipair_config(nat46_remove, 0, config_remove, buflen);
+	for(i = 0; i < nat46->npairs; i++) {
+		char config[buflen];
+		int result = nat46_get_ipair_config(nat46, i, config, buflen);
```

Comment:
(3) because (1) did not increment the reference, the first call at (2) has freed up nat46 pointer, thus the second match will hit here  using nat46 pointing to the free memory...